dali/python/nvidia/dali/_utils/dali_trace.py

```diff
@@ -0,0 +1,184 @@
+            )
+        else:
+            origin_frame_entry = frame_entry
+
+        # We will skip code that is related to internal AutoGraph and conditionals implementation,
+        # leaving us with code produced by transforming user-code
+        # See api.py:converted_call for details on filtering.
+        skip = frame_filter.is_filtered(origin_frame_entry.filename)
+
+        # Detect repeated appearance of function transformed by AG
+        # AutoGraph is wrapping a function call - entry point
+        if is_frame_converted_call(frame_entry):
+            is_ag_function_call_start = True
+            current_function_region = None
+        # It quits to a non-AG code, treat it as normal from now-on
+        if is_frame_call_unconverted(frame_entry):
+            is_ag_function_call_start = False
+            current_function_region = None
+        # We are in the first part of the converted_func call that is not skipped
+        # (as we are in user code, remember the function)
+        if is_ag_function_call_start and not skip:
+            is_ag_function_call_start = False
+            current_function_region = origin_frame_entry
+            skip = True
+            origin_stack_summary.append(origin_frame_entry)
+
+        # User code - not filtered out
+        if not skip:
+            # If we are in the same function region, we replace previous entry so we keep only the
+            # last one
+            assert origin_stack_summary
+            if _is_matching_function(origin_stack_summary[-1], current_function_region):
+                if _collapse_ag_frames:
+                    origin_stack_summary.pop()
+            else:
+                current_function_region = None
+            origin_stack_summary.append(origin_frame_entry)
+        elif not _filter_ag_frames and (
+            len(origin_stack_summary) == 0 or origin_stack_summary[-1] != origin_frame_entry
+        ):
+            origin_stack_summary.append(origin_frame_entry)
+    return origin_stack_summary
+
+
+def extract_stack(skip_bottom_frames=0, skip_top_frames=0):
+    """Extract list of FrameSummary object, optionally skipping the bottom and top ones from
+    the place of the call. If AutoGraph was used, the FrameSummary entries are filtered and remapped
+    back to the user code.
+
+    Returns
+    -------
+    List[FrameSummary]
+    """
+    # Returns a StackSummary which inherits from list, and contains traceback.FrameSummary
+    # objects. Frame summary contains filename, lineno, name and line (string representing context).
+    # -1 so we drop extract_stack frame
+    stack_depth = get_stack_depth()
+    limit = stack_depth - skip_bottom_frames
+    stack_summary = traceback.extract_stack(limit=limit)[: -1 - skip_top_frames]
+
+    # If those are empty, AutoGraph transformations were not used, we can return as is
+    frame_map = get_frame_map()
+    frame_filter = get_frame_filter()
+    if not frame_map and not frame_filter:
```

Comment:
Hmm, without autograph, both are empty.

Reply:
Sure, I just though this way it would rely less on that external assumption and more on when this particular function can work. But I guess it's a bit hypothetical discussion.